### PR TITLE
feat: add styled version of data grid

### PR DESCRIPTION
**Pull request by scomea**
# Description
This change adds a msft styled version of the base data grid component

## Motivation & context
Partner requests.

## Issue type checklist
- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

<!---
Formatting guidelines:

Accepted peer review title format:
<type>: <description>

Example titles:
    chore: add unit tests for all components
    feat: add a border radius to button
    fix: update design system to use 3px border radius

    <type> is required to be one of the following:

        - chore: A change that does not impact distributed packages.
        - fix: A change which fixes an issue.
        - feat: A that adds functionality.

    <description> is required for the CHANGELOG and speaks to what the user gets from this PR:

        - Be concise.
        - Use all lowercase characters. 
        - Use imperative, present tense (e.g. `add` not `adds`.)
        - Do not end your description with a period.
        - Avoid redundant words.

For additional information regarding working on FAST-DNA, check out our documentation site:
https://www.fast.design/docs/en/contributing/working

### CODE REVIEW

File: packages/react/fast-components-styles-msft/src/data-grid/index.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            neutralOutlineRest
+        ),
+    },
+    dataGrid_columnHeader: {
+        height: "100%",
+        "box-sizing": "border-box",
+        overflow: "hidden",
+        padding: format<DesignSystem>("0 {0}", horizontalSpacing(0)),
+        ...ellipsis(),
+        "line-height": "26px",
```
Comment (eljefe223) [inline packages/react/fast-components-styles-msft/src/data-grid/index.ts:52]: Should this be mapped to the design system?

Comment (scomea) [reply packages/react/fast-components-styles-msft/src/data-grid/index.ts]: Hooked it up to height()

**Comment by eljefe223** [comment]
Noticing a focus border around the grid when you click. Probably as simpler as adding `outline: none`

image

**Comment by eljefe223** [comment]
Signing off pending the focus visual gets fixed.

**Pull request merged**